Forge.TreeWalker/src/TreeWalkerSession.cs

```diff
@@ -415,34 +474,43 @@ public async Task<string> VisitNode(string treeNodeKey)
         {
             TreeNode treeNode = this.Schema.Tree[treeNodeKey];
 
-            // Do type-specific behavior.
-            switch (treeNode.Type)
+            if (string.IsNullOrWhiteSpace(this.currentNodeSkipActionContext))
             {
-                case TreeNodeType.Leaf:
-                {
-                    await this.PerformLeafTypeBehavior(treeNode).ConfigureAwait(false);
+                // When this.currentNodeSkipActionContext, it means there is no SkipAction. Just follow the regular route to run Actions.
```

Comment:
>// When this.currentNodeSkipActionContext, it means there is no SkipAction. Just follow the regular route to run Actions. [](start = 16, length = 121)

Do not skip actions when this.currentNodeSkipActionContext is null or whitespace.

Reply:
fixed

---
In reply to: [505832458](https://github.com/microsoft/Forge/pull/32#discussion_r505832458) [](ancestors = 505832458)